lib/collection/src/shards/local_shard/query.rs

```diff
@@ -92,41 +147,38 @@ impl LocalShard {
         'shard: 'query,
     {
         async move {
-            let mut sources: Vec<Vec<ScoredPoint>> = Vec::with_capacity(prefetch.sources.len());
-
-            for source in prefetch.sources.into_iter() {
-                let vec: Vec<Vec<ScoredPoint>> = match source {
-                    PrefetchSource::SearchesIdx(idx) => {
-                        // TODO(universal-query): don't clone, by using something like a hashmap instead of a vec
-                        let scored_searches = core_results.get(idx).cloned().unwrap_or_default();
-                        vec![scored_searches]
-                    }
-                    PrefetchSource::ScrollsIdx(idx) => {
-                        // TODO(universal-query): don't clone, by using something like a hashmap instead of a vec
-                        let scrolled = scrolls.get(idx).cloned().unwrap_or_default();
-                        vec![scrolled]
-                    }
+            let mut search_indices = Vec::new();
+            let mut scroll_indices = Vec::new();
+
+            let mut merged_list = Vec::new();
```

Comment:
Not strictly necessary, but can we make some sane guesstimations here for preallocating capacity?

Reply:
We might preallocate all of them to the length of `merge_plan.sources`.